src/ts/utils/settings.ts

```diff
@@ -0,0 +1,115 @@
+import { browser } from 'webextension-polyfill-ts';
+import { Reducer } from 'redux';
+import { createReducer } from './redux'
+import { getStateFromStorage } from './storage'
+
+
+export interface Textarea {
+    type: 'textarea',
+    id: string,
+    label?: string,
+    initValue?: string,
+    description?: string,
+    onSave?: (value: string) => void,
+}
+
+export type Setting = Textarea
+
+export type Feature = {
+    id: string,
+    name: string,
+    description?: string,
+    shortcuts?: Shortcut[],
+    settings?: Setting[],
+    toggleable?: boolean,
+    toggle?: (active: boolean) => void,
+    reducer?: Reducer
+}
+
+export interface Shortcut {
+    id: string,
+    label: string,
+    initValue: string,
+    placeholder?: string,
+    description?: string,
+    onPress: () => void,
+    onSave?: (shortcut: string) => void
+}
+
+
+
+export const getSetting = async (featureId: string, settingId: string) => {
+    return (await getStateFromStorage())[featureId][settingId]
+}
+
+export const isActive = async (featureId: string) => {
+    return (await getStateFromStorage())[featureId].active
+}
```

Comment:
These 2 functions should be probably exposed under `Settings` object, with fittingly updated names